Source/Dafny/DafnyMain.cs

```diff
@@ -231,9 +232,12 @@ private class IncludeComparer : IComparer<Include> {
       return null; // Success
     }
 
-    public static bool BoogieOnce(ExecutionEngine engine, string baseFile, string moduleName, Microsoft.Boogie.Program boogieProgram, string programId,
-      out PipelineStatistics stats, out PipelineOutcome oc) {
-      programId = (programId ?? "main_program_id") + "_" + moduleName;
+    public static async Task<(bool IsVerified, PipelineOutcome Outcome, PipelineStatistics Statistics)> BoogieOnce(
+      TextWriter output,
+      ExecutionEngine engine, string baseFile,
+      string moduleName,
+      Microsoft.Boogie.Program boogieProgram, string programId) {
+      programId ??= "main_program_id" + "_" + moduleName;
```

Comment:
?

Reply:
This code was already there in the form `programId = (programId ?? "main_program_id") + "_" + moduleName;`